pkg/asset/agent/image/agentimage.go

```diff
@@ -174,6 +176,34 @@ func (a *AgentImage) appendKargs(kargs []byte) error {
 	return nil
 }
 
+// normalizeFilesExtension scans the extracted ISO files and trims
+// the file extensions longer than three chars.
+func (a *AgentImage) normalizeFilesExtension() error {
+	return filepath.WalkDir(a.tmpPath, func(p string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if d.IsDir() {
+			return nil
+		}
+
+		ext := filepath.Ext(p)
+		// ext includes also the dot separator
+		if len(ext) > iso9660Level1ExtLen+1 {
```

Comment:
confused about the `+1` here

Reply:
`filepath.Ext` contains also the `.` char, ie `.json`

Reply:
Oh wow. That's... horrible.